lib/internal/Magento/Framework/Event/ObserverInterface.php

```diff
@@ -10,6 +10,7 @@
 /**
  * Interface \Magento\Framework\Event\ObserverInterface
  *
+ * @api
```

Comment:
It would be good to add a note also that observers creation should be avoid whenever possible in favor of plugins-interceptors.

Reply:
Why? There are plenty of places where an event observer is better suited than a plugin.

Reply:
Mainly because "There can be only one" intention.

The only valid place for event dispatching I heard of are some controllers in MSI project.

All other places in current code which seem better suited by observers is just unrefactored legacy which is doing to much. In code written with plugins in mind there is no need in event dispatching at all.

Reply:
Lol, there is soooo much code that will never be rewritten because it would break so many extensions. Maybe in Magento 3. But for Magento 2 I don't consider that realistic.